lib/libc/string/wcsncmp.c

```diff
@@ -44,16 +44,14 @@ int
 wcsncmp(const wchar_t *s1, const wchar_t *s2, size_t n)
 {
 
-	if (n == 0)
-		return (0);
-	do {
-		if (*s1 != *s2++) {
+	for (; n; --n) {
+		if (*s1 != *s2) {
 			/* XXX assumes wchar_t = int */
-			return (*(const unsigned int *)s1 -
-			    *(const unsigned int *)--s2);
+			return (*s1 - *s2);
 		}
-		if (*s1++ == 0)
+		if (*s1 == L'\0')
 			break;
-	} while (--n != 0);
+		++s1, ++s2;
```

Comment:
these belong in the for loop. That's more consistent with the style and now that they are unconditional, it's easy to move them there.